### PR TITLE
Isolated clusters

### PR DESCRIPTION
**Pull request by majst01**
Related to: https://github.com/fi-ts/proxy-services/issues/4

TODO:

- [ ] test and  merge #33 before

### CODE REVIEW

File: pkg/generator/ignition/generator.go [hunk 1]
```diff
@@ -181,3 +177,89 @@ func getGardenerDecoder() runtime.Decoder {
...
+	renderedContent := `# Generated by os-extension-metal
+imports = ["/etc/containerd/conf.d/*.toml"]
+version = 2
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+`
+	for _, m := range mirrors {
+		for _, of := range m.MirrorOf {
+			renderedContent += fmt.Sprintf(`    [plugins."io.containerd.grpc.v1.cri".registry.mirrors.%q]
```
Comment (majst01) [inline pkg/generator/ignition/generator.go:203]: Missing newline ?

Comment (vknabel) [reply pkg/generator/ignition/generator.go:203]: The newlines are on line 200 and 205 or am I missing something?

File: pkg/generator/ignition/generator_test.go [hunk 2]
```diff
@@ -129,3 +248,11 @@ func TestIgnitionFromOperatingSystemConfig(t *testing.T) {
 		})
 	}
 }
+
+func marshal(t *testing.T, obj runtime.Object) []byte {
```
Comment (majst01) [inline pkg/generator/ignition/generator_test.go:252]:
```suggestion
func mustMarshal(t *testing.T, obj runtime.Object) []byte {
```

File: pkg/generator/ignition/generator.go [hunk 3]
```diff
@@ -137,3 +154,116 @@ func ignitionFromOperatingSystemConfig(config *generator.OperatingSystemConfig)
...
+	}
+	resolveDNS := strings.Join(dnsServers, " ")
+	systemdResolvedConfd := fmt.Sprintf(`# Generated by os-extension-metal
+
+[Resolve]
+DNS=%s
+Domain=~.
+
+`, resolveDNS)
+	resolvConf := "# Generated by gardener-extension-provider-metal\n"
```
Comment (ulrichSchreiner) [inline pkg/generator/ignition/generator.go:229]: Shouldn't this comment say `Generated by os-extension-metal`

File: pkg/generator/ignition/generator.go [hunk 4]
```diff
@@ -137,3 +154,116 @@ func ignitionFromOperatingSystemConfig(config *generator.OperatingSystemConfig)
...
+	for _, ip := range dnsServers {
+		resolvConf += fmt.Sprintf("nameserver %s\n", ip)
+	}
+
```
Comment (ulrichSchreiner) [inline pkg/generator/ignition/generator.go:233]: According to https://man7.org/linux/man-pages/man5/resolv.conf.5.html up to 3 nameservers my be listed

> Up to MAXNS (currently 3, see <resolv.h>) name servers may be listed, one per keyword.

what to do, if there are more than 3 in the `dnsServers` slice?

Comment (vknabel) [reply pkg/generator/ignition/generator.go:233]: Good point! This should ideally be caught by a validating webhook in the gepm.

Comment (majst01) [reply pkg/generator/ignition/generator.go:233]: We should validate this in the gepm networkisolatio validation and prohibit more than 3

Comment (vknabel) [reply pkg/generator/ignition/generator.go:233]: Validation here: https://github.com/metal-stack/gardener-extension-provider-metal/pull/361/commits/ce898b7700dbf8883b7685509310ab5151816160

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**